build.gradle

```diff
@@ -95,31 +99,68 @@ task generateJniInterfaces(type:Exec) {
     environment 'CLASSPATH', sourceSets.main.runtimeClasspath.collect { it.path }.join(':')
     // JNI shared object isn't available at this stage
     environment 'WASMTIME_JNI_LOAD_DISABLED', '1'
-    commandLine(["${project.buildDir}/tmp/javah4x-${javah4xVersion}", 'rust', './wasmtime-jni/src'] + targets)
-    // Manually compare the source (class files) and output (rust sources) to run this task only when any
-    // update is made on classes.
-    outputs.upToDateWhen {
-        def smallestGeneratedTimestamp = targets.collect {
-            new File(project.rootDir, "wasmtime-jni/src/${it.replace(".", "_")}/mod.rs").lastModified()
-        }.min()
-        def largestSourceTimestamp = targets.collect {
-            def path = "classes/java/main/" + it.replace(".", "/") + ".class"
-            new File(project.buildDir, path).lastModified()
-        }.max()
-        smallestGeneratedTimestamp >= largestSourceTimestamp
+    if(jniLibOsClassifier().equals("windows")){
+        File file = new File("./generateJniInterfacesBat.bat")
+        file.createNewFile()
+        file.deleteOnExit()
+        String cmd = ""
+        for(String each:(["cmd","java -cp ${project.buildDir}\\tmp;" +
+                "${project.buildDir}\\tmp\\javah4x;" +
+                "${project.buildDir}\\tmp\\javah4x\\rust;" +
+                "${project.buildDir}\\classes\\java\\main\\io\\github\\kawamuray\\wasmtime;" +
+                "${project.buildDir}\\classes\\java\\main\\io\\github\\kawamuray\\wasmtime\\wasi" +
+                " javah4x.Javah4x", 'rust', '.\\wasmtime-jni\\src'] + targets)){
+            cmd+=(each+" ")
+        }
+        file.write(cmd)
+        commandLine("generateJniInterfacesBat.bat")
+        // Manually compare the source (class files) and output (rust sources) to run this task only when any
+        // update is made on classes.
+        outputs.upToDateWhen {
+            def smallestGeneratedTimestamp = targets.collect {
+                new File(project.rootDir, "wasmtime-jni\\src\\${it.replace(".", "_")}\\mod.rs").lastModified()
```

Comment:
As I commented here: https://github.com/kawamuray/wasmtime-java/issues/1#issuecomment-743691513 , it seems that it's not necessary to use a backslash for paths to support windows platform. Can you try to run the build without changing paths? If it works then the required change will be fairly minimal.

Reply:
OK.